app/Http/Controllers/App/PostController.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+namespace App\Http\Controllers\App;
+
+use App\Http\Controllers\Controller;
+use App\Models\Banner;
+use App\Models\Challenge;
+use App\Models\Comment;
+use App\Models\Post;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Storage;
+use Inertia\Inertia;
+
+class PostController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Inertia\Response
+     */
+    public function index(Request $request)
+    {
+        $posts = Post::with(['author', 'team', 'banner', 'challenge', 'comments', 'comments.author', 'users'])->orderByDesc('updated_at')->get();
+        $posts = $posts->each(function($post){$post->team->image = ($post->team->image) ? Storage::disk('upload')->url($post->team->image) : null;});
+
+        $user = $request->user();
+        $teams = $user->teams;
+        $teams->load('banner');
+
+        return Inertia::render('post/Index', [
+            'posts' => $posts,
+            'banners' => Banner::all()->keyBy('id'),
+            'challenges' => Challenge::all(),
```

Comment:
Here as well, I guess we should limit to the `team.currentChallenges`

Reply:
Here i think it makes sense to allow other challenges. Because some challenges have lasting results which could actually be seen by others. I think mainly the older participants (BDKJ-Vorstand, Freunde und Förderer, Orga-Team etc.) might want to post about those occasions.

Again we should preselect the current challenge and put the teams challenge history on top.